src/Type/Php/BcMathStringOrNullReturnTypeExtension.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\NullType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use function gettype;
+use function strtolower;
+use PhpParser\Node\Expr\UnaryMinus;
+
+class BcMathStringOrNullReturnTypeExtension implements \PHPStan\Type\DynamicFunctionReturnTypeExtension
+{
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return $functionReflection->getName() === 'bcdiv'
+			|| $functionReflection->getName() === 'bcmod'
+			|| $functionReflection->getName() === 'bcpowmod'
+			|| $functionReflection->getName() === 'bcsqrt';
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		$secondArgument = $functionCall->args[1]->value->value;
+		$functionName = strtolower($functionReflection->getName());
+
+		if ($secondArgument === '0'
```

Comment:
In this way, if the divider is variable, then we still say it always returns a string, which is not correct:

```php

function divideThreeBy(string $divider): string
{
    retrun bcdiv('3', $divider);
}

```
divideThreeBy('0'); Will still make this nullable

Reply:
The same goes for non numeric strings: https://3v4l.org/7Fsgc

Reply:
Ohh. Is it possible to resolve the value? Then return UnionType if not possible to resolve it?